docs/data/date-pickers/custom-field/MaterialV6FieldWrapped.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import MuiTextField, { TextFieldProps } from '@mui/material/TextField';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateField } from '@mui/x-date-pickers/DateField';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+
+const TextField = React.forwardRef(
+  (props: TextFieldProps, ref: React.Ref<HTMLDivElement>) => (
+    <MuiTextField {...props} ref={ref} size="small" />
```

Comment:
> This example is way easier to achieve via slotProps

In the doc yes, is a reall app not always
The use case I'm covering here is people that have a custom `TextField` that they use everywhere is their app (in the pickers, but also in the autocomplete or even called directly).
They could use `slotProps`, but they already have a `TextField` component ready with all the customizations.

I could make the customization more advanced to make clear the `slotProps` is not an option if you think that it clarifies the purpose of this demo.

Reply:
> I'm not sure we should be giving such examples, as I fear that users might start preferring usage of slots for the slightest modification. 

IMHO it's a very valid use case if people are using this custom component elsewhere (because it allows collocation) 
But it's a terrible use case if they just use it on some picker

Maybe we could add an `:::info` to explain when this approach is recommended

Reply:
Yes, maybe a callout would be the best idea. We wouldn't pollute the demo with too much information.
Your point regarding the usage of a `slot` to re-use the same `TextField` users have for the rest of the application sounds valid.
However, I'd be curious to see, how many users do it, instead of using theme overrides. 🤔

Reply:
> However, I'd be curious to see, how many users do it, instead of using theme overrides. 

I don't have a lot of data to back that up to be honest.
Last time I checked an app that was using the pickers, it had this pattern.
I think that it's pretty widely used, and even the default way of customizing for people that don't dive into the features specific to MUI (theme augmentation is not something available in every framework, creating a wrapper is, so I would expect someone coming from another framework to often start with it).